pkg/cluster/deploystorage_resources.go

```diff
@@ -76,12 +76,142 @@ func (m *manager) denyAssignments(clusterSPObjectID string) *arm.Resource {
 	}
 }
 
-func (m *manager) clusterServicePrincipalRBAC(clusterSPObjectID string) *arm.Resource {
-	return rbac.ResourceGroupRoleAssignmentWithName(
-		rbac.RoleContributor,
-		"'"+clusterSPObjectID+"'",
-		"guid(resourceGroup().id, 'SP / Contributor')",
-	)
+func (m *manager) clusterServicePrincipalRBAC(clusterSPObjectID string) []*arm.Resource {
+	if feature.IsRegisteredForFeature(m.subscriptionDoc.Subscription.Properties, "Microsoft.RedHatOpenShift/RedHatEngineering") {
+		m.log.Printf("RedHat Engineering flag detected - deploying CSP role")
+		//RedHatEngineering feature flag
+		roles := []*arm.Resource{
+			rbac.ResourceRoleDefinition("guid(resourceGroup().id, 'Azure Red Hat OpenShift autogenerated RG CSP')",
+				[]mgmtauthorization.Permission{
+					{
+						Actions: &[]string{
+							"Microsoft.Network/applicationSecurityGroups/read",
```

Comment:
@Makdaam please could you schedule a call to review these and get these straight before we merge this?

Reply:
@Makdaam , Did the call happen? What was the outcome?

Reply:
The call happened, I have action items, putting the PR back in draft until I apply the changes.

Reply:
Simplified the actions/notactions of the new role.
Added architecture version dependent notActions, and additional deny assignment rules to the VSTS workitem to address after this PR.